app/views/contexts/section_summary_context.py

```diff
@@ -113,18 +113,37 @@ def _title_for_location(self, location):
         )
         return title
 
-    def _custom_summary_elements(self, section_summary, current_location, section):
+    def _custom_summary_elements(
+        self, section_summary, current_location, section, routing_path
+    ):
         for summary_element in section_summary:
             if summary_element["type"] == "List":
                 yield self._list_summary_element(
-                    summary_element, current_location, section
+                    summary_element, current_location, section, routing_path
                 )
 
-    def _list_summary_element(self, summary, current_location, section) -> Mapping:
+    def _list_summary_element(
+        self, summary, current_location, section, routing_path
+    ) -> Mapping:
         current_list = self._list_store[summary["for_list"]]
-        list_collector_block = self._schema.get_list_collector_for_list(
-            section, for_list=summary["for_list"]
-        )
+
+        list_collector_blocks_on_path = [
```

Comment:
A few things here:

1) We should get a variable that contains all of the list collector blocks - `list_collector_blocks`
2) Using `list_collector_blocks`, create `list_collector_blocks_on_path`, which is already being done here
3) The test on line 167 can then be negated:
```
if not list_collector_block:
    list_collector_block = list_collector_blocks[0]
```

that way, there will always be a list context available, even when there are no list collector blocks on the path but primary person is added.

To test for the scenario, we probably need a new schema. It should have a PrimaryPersonListCollector, and a single ListCollector, with a ListCollectorDrivingQuestion. The test should determine primary person is displayed correctly on the section summary, when there is a primary person added, but no list collectors on the path (i.e. the negative ListCollectorDrivingQuestion is selected)

Reply:
@MebinAbraham does the above fix the issue you raised?

Reply:
Looks good.

Reply:
Test schema added with that specific scenario tested. Changes James recommended added. There is this side effect now where I needed to add edit block to list context as normal primary person edit block id is not used to resolve change link(in this new test scenario)🤔.